x-pack/plugins/security_solution/server/lib/detection_engine/signals/rule_executor.ts

```diff
@@ -0,0 +1,684 @@
+    updatedBy,
+    enabled,
+    schedule: { interval },
+    throttle,
+  } = savedObject.attributes;
+  const updatedAt = savedObject.updated_at ?? '';
+  const refresh = actions.length ? 'wait_for' : false;
+  const buildRuleMessage = buildRuleMessageFactory({
+    id: alertId,
+    ruleId,
+    name,
+    index: outputIndex,
+  });
+
+  logger.debug(buildRuleMessage('[+] Starting Signal Rule execution'));
+  logger.debug(buildRuleMessage(`interval: ${interval}`));
+  let wroteWarningStatus = false;
+  await ruleStatusService.goingToRun();
+
+  // check if rule has permissions to access given index pattern
+  // move this collection of lines into a function in utils
+  // so that we can use it in create rules route, bulk, etc.
+  try {
+    if (!isEmpty(index)) {
+      const hasTimestampOverride = timestampOverride != null && !isEmpty(timestampOverride);
+      const inputIndices = await getInputIndex(services, version, index);
+      const [privileges, timestampFieldCaps] = await Promise.all([
+        checkPrivileges(services, inputIndices),
+        services.scopedClusterClient.fieldCaps({
+          index,
+          fields: hasTimestampOverride
+            ? ['@timestamp', timestampOverride as string]
+            : ['@timestamp'],
+          include_unmapped: true,
+        }),
+      ]);
+
+      wroteWarningStatus = await flow(
+        () =>
+          tryCatch(
+            () => hasReadIndexPrivileges(privileges, logger, buildRuleMessage, ruleStatusService),
+            toError
+          ),
+        chain((wroteStatus) =>
+          tryCatch(
+            () =>
+              hasTimestampFields(
+                wroteStatus,
+                hasTimestampOverride ? (timestampOverride as string) : '@timestamp',
+                timestampFieldCaps,
+                inputIndices,
+                ruleStatusService,
+                logger,
+                buildRuleMessage
+              ),
+            toError
+          )
+        ),
+        toPromise
+      )();
+    }
+  } catch (exc) {
+    logger.error(buildRuleMessage(`Check privileges failed to execute ${exc}`));
+  }
+
+  const gap = getGapBetweenRuns({ previousStartedAt, interval, from, to });
+  if (gap != null && gap.asMilliseconds() > 0) {
+    const fromUnit = from[from.length - 1];
+    const { ratio } = getGapMaxCatchupRatio({
+      logger,
+      buildRuleMessage,
+      previousStartedAt,
+      ruleParamsFrom: from,
+      interval,
+      unit: fromUnit,
+    });
+    if (ratio && ratio >= MAX_RULE_GAP_RATIO) {
+      const gapString = gap.humanize();
+      const gapMessage = buildRuleMessage(
+        `${gapString} (${gap.asMilliseconds()}ms) has passed since last rule execution, and signals may have been missed.`,
+        'Consider increasing your look behind time or adding more Kibana instances.'
+      );
+      logger.warn(gapMessage);
+
+      hasError = true;
+      await ruleStatusService.error(gapMessage, { gap: gapString });
+    }
+  }
+  try {
```

Comment:
Can you justify your use of a `try` catch block here? It seems to cover errors in a large amount of code. Could it catch non-application specific errors (e.g. type errors, range errors, etc)?